crates/starknet/src/starknet/add_declare_transaction.rs

```diff
@@ -13,34 +14,35 @@ pub fn add_declare_transaction_v2(
     broadcasted_declare_transaction: BroadcastedDeclareTransactionV2,
 ) -> DevnetResult<(TransactionHash, ClassHash)> {
     if broadcasted_declare_transaction.common.max_fee.0 == 0 {
-        return Err(Error::TransactionError(
-            starknet_in_rust::transaction::error::TransactionError::FeeError(
-                "For declare transaction version 2, max fee cannot be 0".to_string(),
-            ),
-        ));
+        return Err(Error::FeeError {
+            reason: "For declare transaction version 2, max fee cannot be 0".to_string(),
+        });
     }
 
-    let sir_declare_transaction =
-        broadcasted_declare_transaction.create_sir_declare(starknet.config.chain_id.to_felt())?;
+    let blockifier_declare_transaction =
+        broadcasted_declare_transaction.create_blockifier_declare(starknet.chain_id().to_felt())?;
 
-    let transaction_hash = sir_declare_transaction.hash_value.clone().into();
-    let class_hash: ClassHash = sir_declare_transaction.sierra_class_hash.clone().into();
+    let transaction_hash = blockifier_declare_transaction.tx_hash().0.into();
```

Comment:
starknet_api use newtype idiom for many types. This means we will have many converters. Lets make a separate issue for this.

Reply:
Ok, I wasn't aware of [the newtype idiom](https://doc.rust-lang.org/rust-by-example/generics/new_types.html). Since this is a common idiom and looks recommended, I'm ok with keeping it this way.